### PR TITLE
dagster-external docker asset

### PR DESCRIPTION
**Pull request by alangenfeld**
extend dagster-external to docker 

## How I Tested These Changes

added test

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15820** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15820" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15820?utm_source=stack-comment).

**Comment by smackesey** [comment]
What's the standard here? We could probably iterate this to get it cleaner (working out a solution to overriding the io mode decorators that you are adding additional logic to in run), but IMO it makes sense to land as is as soon as tests pass for easier collaborative iteration.

**Review by schrockn — APPROVED** [review]
@smackesey re: our standard I think we should bias towards landing stuff in master and iterating

File: python_modules/libraries/dagster-docker/dagster_docker_tests/external/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+services:
+  test:
+    image: cool_data_image
+    build:
+      context: .
+      additional_contexts:
+        dagster-external: ../../../../dagster-external
```
[the remark this answers is not in the document]
Comment (schrockn) [reply python_modules/libraries/dagster-docker/dagster_docker_tests/external/docker-compose.yml:8]: ?

[the remark this answers is not in the document]
Comment (alangenfeld) [reply python_modules/libraries/dagster-docker/dagster_docker_tests/external/docker-compose.yml:8]: using inline comment to have a permalink to share to point at specific code

File: python_modules/libraries/dagster-docker/dagster_docker_tests/test_external_asset.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    get_test_project_docker_image,
+)
+
+
+@pytest.mark.parametrize(
+    ["input_spec", "output_spec"],
+    [
+        *(
+            itertools.product(("file", "socket"), ("file", "socket"))
+            if not IS_BUILDKITE
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-docker/dagster_docker_tests/test_external_asset.py:20]: getting sockets working in buildkite deemed not worth it - quite challenging

**Pull request merged**